### DISCUSSION TITLE
How to avoid auto adjust of top annotations when combining heatmaps

### DISCUSSION THREAD

**Discussion by nadluru** [jokergoo/ComplexHeatmap#1019] [Q&A]
Dear Zuguang, 

Firstly, thank you so much for developing such a wonderful package with very helpful documentation!

I am trying to combine two heatmaps using draw. I was hoping I could combine them without auto adjusting the heights of the top annotations in the two heatmaps. Is this possible to achieve with some settings? I wasn't able to figure this out by looking at the documentation and trying a few different things. If one of the heatmaps has a top and the other bottom annotation, then they are not scaled but if both have the top annotations then they are getting re-scaled as seen in the images below.

![image](https://user-images.githubusercontent.com/5951009/205197909-77bfce53-392b-4f32-a351-f506459b21e4.png)

![image](https://user-images.githubusercontent.com/5951009/205197947-548cd8a5-8ab0-4558-a307-2dc24ad7e9ba.png)

Could you kindly please help me with this? Please let me know if you would advise me to provide more information on my request.

Thank you so much!
Sincerely,
Nagesh

**Comment by jokergoo** [comment]
Can you also provide your code?

**Reply by nadluru** [reply]
Dear Zuguang,

Thank you so much for your attention. I needed to clean up the code slightly for better readability. I am attaching both [code and data](https://github.com/jokergoo/ComplexHeatmap/files/10187451/CombiningHeatmapsExample.zip).


Please let me know if I could provide any additional information.

Thank you so much for your help!
Sincerely,
Nagesh

**Reply by jokergoo** [reply]
Sorry for the late reply. 

I think in the second plot, you just need to set a same `ylim = c(0, 30)` to both barplot annotations

**Reply by nadluru** [reply]
Dear Zuguang,

Thank you for the suggestion. No worries about the delay. I sincerely apologize but I was not sure how to use `ylim` option for the purpose. My main intention is to scale the max height of the bar in the second plot so that we can compare it with the first plot when combining them using `draw`.

The top annotation in the first plot has a bar that is 2 cm long, set as below.

`
top_annotation = HeatmapAnnotation(dist1 = anno_barplot(
       ...,
        height = unit(2, "cm")
      ), ...)
`

I would like the bar in the top annotation for second plot to be scaled by the max values as 2(max2/max1) cm long, set as below.

`
top_annotation = HeatmapAnnotation(dist1 = anno_barplot(
       ...,
        height = unit((2*(thm2 %>% colSums(.) %>% max)) / (thm1 %>% colSums(.) %>% max), "cm")
      ), ...)
`

I would like to note this is not an issue when using a bottom annotation in second plot. My sense is that it has to do something with `auto_adjust` feature when combining different plots.

Is your suggestion to use a different approach that involves `ylim` and not `height`? If so, could you kindly send a sample code or pointer to accomplish this?

Thank you so much for your help!
Sincerely,
Nagesh

**Reply by jokergoo** [reply]
In your case, the two barplots are using the same measure. In more general cases, ComplexHeatmap will not know whether they are in the same measure or not, and it simply treats them as two annotations.

**Reply by nadluru** [reply]
Dear Zuguang,

I apologize but I am trying to fully interpret your comment. Is your suggestion to not combine the heat maps using `draw` for this purpose?

Thank you for your time and consideration of my request on this thread!

Sincerely,
Nagesh
